oauthproxy.go

```diff
@@ -109,27 +109,34 @@ func (u *UpstreamProxy) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 
 }
 
-func NewReverseProxy(target *url.URL, upstreamFlush time.Duration, rootCAs []string) (*httputil.ReverseProxy, error) {
+func NewReverseProxy(target *url.URL, opts *Options) (*httputil.ReverseProxy, error) {
 	proxy := httputil.NewSingleHostReverseProxy(target)
-	proxy.FlushInterval = upstreamFlush
+	proxy.FlushInterval = opts.UpstreamFlush
 
-	transport := &http.Transport{
-		MaxIdleConnsPerHost: 500,
-		IdleConnTimeout:     1 * time.Minute,
-	}
-	if len(rootCAs) > 0 {
-		pool, err := util.GetCertPool(rootCAs, false)
+	// Inherit default transport options from Go's stdlib
+	transport := http.DefaultTransport.(*http.Transport).Clone()
+
+	transport.MaxIdleConnsPerHost = 500
+	transport.IdleConnTimeout = 1 * time.Minute
+
+	// Change default duration for waiting for an upstream response
+	transport.ResponseHeaderTimeout = opts.Timeout
```

Comment:
the previous commit does not do anything without this line, does it?

Reply:
Allow me to change the order of things a little here.

Reply:
Done. Thank you!